server/images.go

```diff
@@ -491,31 +491,18 @@ func CreateModel(ctx context.Context, name, modelFileDir, quantization string, c
 		}
 	}
 
-	unref := make(map[string]struct{})
-	if manifest, _, err := GetManifest(ParseModelPath(name)); err == nil {
-		for _, layer := range manifest.Layers {
-			if !slices.Contains(digests, layer.Digest) {
-				unref[layer.Digest] = struct{}{}
-			}
-		}
-
-		if manifest.Config.Digest != layer.Digest {
-			unref[manifest.Config.Digest] = struct{}{}
+	if os.Getenv("OLLAMA_NOPRUNE") == "" {
+		fn(api.ProgressResponse{Status: "removing unused layers"})
+		if manifest, err := ParseNamedManifest(name); err == nil {
+			_ = manifest.Remove()
 		}
 	}
 
 	fn(api.ProgressResponse{Status: "writing manifest"})
-	if err := WriteManifest(name, layer, layers); err != nil {
+	if _, err := NewManifest(name, layer, layers); err != nil {
```

Comment:
Write was far more clear. This has side-effects, on disk, and is not simply instantiating some new value in memory. I use Put for this in my code, but can change it later. For now I would keep Write to avoid surprises.